tests/test_repin_vendor.py

```
@@ -0,0 +1,512 @@
+
+
+def _current_sha_kw(*, vendor: object) -> str:
+    """Return the vendor's current pin SHA.
+
+    Args:
+        vendor: Registry vendor.
+
+    Returns:
+        The existing pin SHA.
+    """
+    return str(getattr(vendor, "sha"))
```

Ruff (0.16.2)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)